File: app/src/main/java/com/zaclimon/aceiptv/main/MainActivity.java
package com.zaclimon.aceiptv.main;

import android.app.Activity;
import android.app.FragmentTransaction;
import android.content.ContentResolver;
import android.content.Intent;
import android.media.tv.TvContract;
import android.os.Bundle;

import com.zaclimon.aceiptv.R;
import com.zaclimon.aceiptv.auth.AuthActivityTv;
import com.zaclimon.aceiptv.util.AceChannelUtil;
import com.zaclimon.aceiptv.util.ActivityUtil;


/**
 * First activity upon opening the application. Decides of the layouts to use and if
 * a sign-in is required.
 *
 * @author zaclimon
 * Creation date: 20/06/17
 */

public class MainActivity extends Activity {

    private final int AUTH_REQUEST = 0;

    /**
     * {@inheritDoc}
     */
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        if (ActivityUtil.isUsernamePasswordEmpty(this)) {

            ContentResolver contentResolver = getContentResolver();
            Intent intent = new Intent(this, AuthActivityTv.class);
            String inputId = TvContract.buildInputId(AceChannelUtil.TV_INPUT_SERVICE_COMPONENT);

            // Delete all channels in case of where the user data has been cleared.
            contentResolver.delete(TvContract.buildChannelsUriForInput(inputId), null, null);
            startActivityForResult(intent, AUTH_REQUEST);
        } else {
            configureLayout();
        }
    }

    /**
     * Adds the correct layout based on the display type
     */
    private void configureLayout() {
        FragmentTransaction fragmentTransaction = getFragmentManager().beginTransaction();

        if (ActivityUtil.isTvMode(this)) {
            fragmentTransaction.add(R.id.activity_fragment_holder, new MainTvFragment());
        }

        fragmentTransaction.commit();
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {

        if (requestCode == AUTH_REQUEST && resultCode == Activity.RESULT_OK) {
            configureLayout();
        } else {
            finish();
        }

    }
}

File: app/src/main/java/com/zaclimon/tsutaeru/ui/main/MainActivity.kt
package com.zaclimon.tsutaeru.ui.main

import android.content.Context
import android.content.Intent
import android.media.tv.TvContract
import android.os.AsyncTask
import android.os.Bundle
import androidx.fragment.app.FragmentActivity
import android.widget.Toast
import com.zaclimon.tsutaeru.R
import com.zaclimon.tsutaeru.ui.auth.AuthActivityTv
import com.zaclimon.tsutaeru.util.ActivityUtil
import com.zaclimon.tsutaeru.util.Constants
import java.lang.ref.WeakReference

/**
 * First activity upon opening the application. Decides of the layouts to use and if
 * a sign-in is required.
 *
 * @author zaclimon
 */
class MainActivity : FragmentActivity() {

    companion object {
        private const val REQUEST_AUTH: Int = 0
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        if (ActivityUtil.areCredentialsEmpty(this) && ActivityUtil.isTvMode(this)) {
            val intent = Intent(this, AuthActivityTv::class.java)
            AsyncDeleteChannels(this).execute()
            startActivityForResult(intent, REQUEST_AUTH)
        } else {
            configureLayout()
        }
    }

    private fun configureLayout() {

        if (ActivityUtil.isTvMode(this)) {
            val transaction = supportFragmentManager.beginTransaction()
            transaction.add(R.id.activity_fragment_holder, MainTvFragment())
            transaction.commit()
        } else {
            Toast.makeText(this, R.string.android_tv_only_toast, Toast.LENGTH_SHORT).show()
            finish()
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        if (requestCode == REQUEST_AUTH && resultCode == androidx.fragment.app.FragmentActivity.RESULT_OK) {
            // Recreate the activity so we won't get a state loss after onSaveInstanceState is called.
            recreate()
        } else {
            finish()
        }
    }

    /**
     * [AsyncTask] that deletes all channels in case of where the user data might have been cleared.
     *
     * @author zaclimon
     */
    private class AsyncDeleteChannels(context: Context) : AsyncTask<Void?, Void?, Void?>() {

        private val asyncContext = WeakReference(context)

        override fun doInBackground(vararg params: Void?): Void? {
            val context = asyncContext.get()
            val inputId = TvContract.buildInputId(Constants.TV_INPUT_SERVICE_COMPONENT)
            val channelUri = TvContract.buildChannelsUriForInput(inputId)
            val contentResolver = context?.contentResolver
            channelUri?.let {
                contentResolver?.delete(it, null, null)
            }
            return null
        }
    }
}